app/js/arethusa.core/services/configurator.js

```diff
@@ -380,51 +356,20 @@ angular.module('arethusa.core').service('configurator', [
       }
 
       setGlobalDefaults(obj);
-    };
-
-    function setGlobalDefaults(obj) {
-      angular.forEach(getGlobalDefaults(), function(value, key) {
-        // Explicitly ask for undefined, as a false value can be a
-        // valid configuration seting!
-        if (obj[key] === undefined) {
-          obj[key] = value;
-        }
-      });
-    }
-
-    var globalDefaults = {
-      'mode' : 'editor'
-    };
-    function getGlobalDefaults() {
-      var customDefaults = getGlobalCustomDefaults();
-      var routeDefaults  = getGlobalDefaultsFromRoute();
-      return angular.extend({}, globalDefaults, customDefaults, routeDefaults);
-    }
-
-    function getGlobalCustomDefaults() {
-      return self.configuration.main.globalDefaults || {};
-    }
-
-    var routeParams = ['mode'];
-    function getGlobalDefaultsFromRoute() {
-      return arethusaUtil.inject({}, routeParams, function(memo, param) {
-        var value = $location.search()[param];
-        if (value) memo[param] = value;
-      });
-    }
-
-    /**
-     * @ngdoc function
-     * @name arethusa.core.configurator#mode
-     * @methodOf arethusa.core.configurator
-     *
-     * @description
-     * Getter to read the current global mode of the application.
-     *
-     * @returns {String} The current mode, e.g. `'editor'` or `'viewer'`.
-     */
-    this.mode = function() {
-      return getGlobalDefaults().mode;
+      var uPCached;
```

Comment:
@fbaumgardt should the declaration of uPCached this be moved back out into a class variable? I think as you currently have it, the userPreferences will never actually get cached because the var is redeclared each time delegateConf is called.